src/components/ProductDescription/QuantityTextField.tsx

```diff
@@ -15,39 +15,42 @@ export const QuantityTextField: React.FC<QuantityTextFieldProps> = ({
   onQuantityChange,
   hideErrors,
 }: QuantityTextFieldProps) => {
-  const [tempQuantity, setTempQuantity] = useState<number | string>(quantity);
+  const [tempQuantity, setTempQuantity] = useState<string>(quantity.toString());
   const [isTooMuch, setIsTooMuch] = useState(false);
 
   const handleBlurQuantityInput = () => {
-    const newQuantity =
-      typeof tempQuantity === "number"
-        ? tempQuantity
-        : parseInt(tempQuantity, 10);
-    const notEnoughQuantity = isNaN(newQuantity) || newQuantity <= 0;
-    if (notEnoughQuantity) {
-      setTempQuantity(quantity);
+    let newQuantity = parseInt(tempQuantity, 10);
```

Comment:
Isint that same logic twice? Why not move it as util or atom component?

Reply:
It's a bit different in cart page, where it sets input to max value if quantity exceeds.